singleheader/CMakeLists.txt

```diff
@@ -17,6 +17,36 @@ set(SINGLEHEADER_REPO_FILES
 )
 set_source_files_properties(${SINGLEHEADER_FILES} PROPERTIES GENERATED TRUE)
 
+#
+# Build the library
+#
+if(SIMDJSON_BUILD_STATIC)
+  MESSAGE( STATUS "Building a static library." )
+  add_library(simdjson STATIC "")
+else()
+  MESSAGE( STATUS "Building a dynamic library." )
+  add_library(simdjson SHARED "")
+  target_compile_definitions(simdjson INTERFACE SIMDJSON_USING_LIBRARY=1)
+  if(MSVC)
+    MESSAGE( STATUS "Building a Windows DLL using Visual Studio, exporting all symbols automatically." )
+    set_target_properties(simdjson PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS 1)
+  endif()
+endif()
+
+target_include_directories(simdjson PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
```

Comment:
Something I don't understand that maybe you could explain: why do we include both of these directories, rather than just one or the other?

Reply:
I ask out of curiosity rather than concern: I've seen this pattern before, I just don't understand it. :)

Reply:
INSTALL_INTERFACE is used for stating the requirements for the consumers. Since after importing simdjson in CMake they will not be using the current include/ directory but <prefix>/include

Though that wasn't causing any problems before since install(TARGETS) command with INCLUDES DESTINATION parameter act more-or-less the same but I've changed since this one seemed more common